ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -3603,7 +3603,20 @@ private Operator genFilterPlan(ASTNode searchCond, QB qb, Operator input,
          * push filters only for this QBJoinTree. Child QBJoinTrees have already been handled.
          */
         pushJoinFilters(qb, joinTree, aliasToOpInfo, false);
-        input = genJoinOperator(qbSQ, joinTree, aliasToOpInfo, input);
+        boolean notInCheck = (subQuery.getNotInCheck() != null)?true : false;
+        if(notInCheck && !qb.isMultiDestQuery()){
```

Comment:
There is already a code flow at line 3590 that handles notInCheck. See if this code flow can be combined there.

Reply:
The code block with the notInCheck at that line, handles the not in operator such that if subquery table.col_referenced has a null value, output will be empty. The code block added in this PR aligns with the behaviour of queries with "IN"  operator, where the null not in case is handled during join time by adding a predicate.